pkgs/applications/window-managers/i3/i3-resurrect.nix

```diff
@@ -0,0 +1,22 @@
+{ lib, buildPythonApplication, click, i3ipc, psutil, natsort, fetchPypi, xprop, xdotool, importlib-metadata }:
+
+buildPythonApplication rec {
+  pname = "i3-resurrect";
+  version = "1.4.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0h181frdwpqfj9agw43qgicdvzv1i7xwky0vs0ksd8h18qxqp4hr";
+  };
+
+  propagatedBuildInputs = [ click psutil xprop natsort i3ipc xdotool importlib-metadata ];
+  doCheck = false;
```

Comment:
Can you please comment why you disabled the tests.

Reply:
When I remove this line, the test tries to write to ``/homeless-shelter/``.
When I add:
```nix
  preCheck = ''
    export HOME=$TMPDIR
  '';
```
I get:
```
checking outputs of '/nix/store/1hixkwnkr7mlv9vppb7dqw3nkz9p6knp-i3-resurrect-1.4.3.drv'...
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
unpacking sources
unpacking source archive /nix/store/k6dymb0m1lipasmgl6lb7k13vwkpjijg-i3-resurrect-1.4.3.tar.gz
source root is i3-resurrect-1.4.3
setting SOURCE_DATE_EPOCH to timestamp 1584537389 of file i3-resurrect-1.4.3/setup.cfg
patching sources
configuring
no configure script, doing nothing
building
Executing setuptoolsBuildPhase
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/i3_resurrect
copying i3_resurrect/util.py -> build/lib/i3_resurrect
copying i3_resurrect/config.py -> build/lib/i3_resurrect
copying i3_resurrect/main.py -> build/lib/i3_resurrect
copying i3_resurrect/layout.py -> build/lib/i3_resurrect
copying i3_resurrect/__init__.py -> build/lib/i3_resurrect
copying i3_resurrect/treeutils.py -> build/lib/i3_resurrect
copying i3_resurrect/programs.py -> build/lib/i3_resurrect
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/i3_resurrect
copying build/lib/i3_resurrect/util.py -> build/bdist.linux-x86_64/wheel/i3_resurrect
copying build/lib/i3_resurrect/config.py -> build/bdist.linux-x86_64/wheel/i3_resurrect
copying build/lib/i3_resurrect/main.py -> build/bdist.linux-x86_64/wheel/i3_resurrect
copying build/lib/i3_resurrect/layout.py -> build/bdist.linux-x86_64/wheel/i3_resurrect
copying build/lib/i3_resurrect/__init__.py -> build/bdist.linux-x86_64/wheel/i3_resurrect
copying build/lib/i3_resurrect/treeutils.py -> build/bdist.linux-x86_64/wheel/i3_resurrect
copying build/lib/i3_resurrect/programs.py -> build/bdist.linux-x86_64/wheel/i3_resurrect
running install_egg_info
running egg_info
writing i3_resurrect.egg-info/PKG-INFO
writing dependency_links to i3_resurrect.egg-info/dependency_links.txt
writing entry points to i3_resurrect.egg-info/entry_points.txt
writing requirements to i3_resurrect.egg-info/requires.txt
writing top-level names to i3_resurrect.egg-info/top_level.txt
reading manifest file 'i3_resurrect.egg-info/SOURCES.txt'
writing manifest file 'i3_resurrect.egg-info/SOURCES.txt'
Copying i3_resurrect.egg-info to build/bdist.linux-x86_64/wheel/i3_resurrect-1.4.3-py3.8.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/i3_resurrect-1.4.3.dist-info/WHEEL
creating 'dist/i3_resurrect-1.4.3-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'i3_resurrect/__init__.py'
adding 'i3_resurrect/config.py'
adding 'i3_resurrect/layout.py'
adding 'i3_resurrect/main.py'
adding 'i3_resurrect/programs.py'
adding 'i3_resurrect/treeutils.py'
adding 'i3_resurrect/util.py'
adding 'i3_resurrect-1.4.3.dist-info/METADATA'
adding 'i3_resurrect-1.4.3.dist-info/WHEEL'
adding 'i3_resurrect-1.4.3.dist-info/entry_points.txt'
adding 'i3_resurrect-1.4.3.dist-info/top_level.txt'
adding 'i3_resurrect-1.4.3.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Finished executing setuptoolsBuildPhase
installing
Executing pipInstallPhase
/build/i3-resurrect-1.4.3/dist /build/i3-resurrect-1.4.3
DEPRECATION: The -b/--build/--build-dir/--build-directory option is deprecated. pip 20.3 will remove support for this functionality. A possible replacement is use the TMPDIR/TEMP/TMP environment variable, possibly combined with --no-clean. You can find discussion regarding this at https://github.com/pypa/pip/issues/8333.
Processing ./i3_resurrect-1.4.3-py3-none-any.whl
Requirement already satisfied: Click in /nix/store/nbn1bcn2h42mlsmdfx2c7q47c1hf19zj-python3.8-click-7.1.2/lib/python3.8/site-packages (from i3-resurrect==1.4.3) (7.1.2)
Requirement already satisfied: psutil in /nix/store/jbl5w80cjd2pn315655baf6l09p7xnil-python3.8-psutil-5.7.3/lib/python3.8/site-packages (from i3-resurrect==1.4.3) (5.7.3)
Requirement already satisfied: natsort in /nix/store/znwwlrb260vgjwg3n1xrydj0l0gi7lrv-python3.8-natsort-7.0.1/lib/python3.8/site-packages (from i3-resurrect==1.4.3) (7.0.1)
Requirement already satisfied: i3ipc in /nix/store/5pb6dlrxyznczsjqi16i1h57ccgbn6dn-python3.8-i3ipc-2.2.1/lib/python3.8/site-packages (from i3-resurrect==1.4.3) (2.2.1)
Requirement already satisfied: python-xlib in /nix/store/26byrinql4ahn7s28b4v08yinzdhv2gw-python3.8-xlib-0.28/lib/python3.8/site-packages (from i3ipc->i3-resurrect==1.4.3) (0.28)
Requirement already satisfied: six>=1.10.0 in /nix/store/rl4s0fhwhx3pmr67si1y7djhqln7g97w-python3.8-six-1.15.0/lib/python3.8/site-packages (from python-xlib->i3ipc->i3-resurrect==1.4.3) (1.15.0)
Installing collected packages: i3-resurrect
  WARNING: The script i3-resurrect is installed in '/nix/store/j28cl8fffjbpiqkkwxx7f970mlf4369s-i3-resurrect-1.4.3/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed i3-resurrect-1.4.3
/build/i3-resurrect-1.4.3
Finished executing pipInstallPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/j28cl8fffjbpiqkkwxx7f970mlf4369s-i3-resurrect-1.4.3
strip is /nix/store/p3kn26g5nhmx6spn37ar3mn0xjbzks3g-binutils-2.31.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/j28cl8fffjbpiqkkwxx7f970mlf4369s-i3-resurrect-1.4.3/lib  /nix/store/j28cl8fffjbpiqkkwxx7f970mlf4369s-i3-resurrect-1.4.3/bin
patching script interpreter paths in /nix/store/j28cl8fffjbpiqkkwxx7f970mlf4369s-i3-resurrect-1.4.3
checking for references to /build/ in /nix/store/j28cl8fffjbpiqkkwxx7f970mlf4369s-i3-resurrect-1.4.3...
Rewriting #!/nix/store/5aq2ix4vmk8bxs20ddqi48vc4bnk8zr5-python3-3.8.6/bin/python3.8 to #!/nix/store/5aq2ix4vmk8bxs20ddqi48vc4bnk8zr5-python3-3.8.6
wrapping `/nix/store/j28cl8fffjbpiqkkwxx7f970mlf4369s-i3-resurrect-1.4.3/bin/i3-resurrect'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests
no Makefile or custom installCheckPhase, doing nothing
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
setuptoolsCheckPhase
Executing setuptoolsCheckPhase
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
writing i3_resurrect.egg-info/PKG-INFO
writing dependency_links to i3_resurrect.egg-info/dependency_links.txt
writing entry points to i3_resurrect.egg-info/entry_points.txt
writing requirements to i3_resurrect.egg-info/requires.txt
writing top-level names to i3_resurrect.egg-info/top_level.txt
reading manifest file 'i3_resurrect.egg-info/SOURCES.txt'
writing manifest file 'i3_resurrect.egg-info/SOURCES.txt'
running build_ext

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
Finished executing setuptoolsCheckPhase
/nix/store/j28cl8fffjbpiqkkwxx7f970mlf4369s-i3-resurrect-1.4.3
```

Shall I change the PR?
I'm asking  cause the output mentions:
```
Ran 0 tests in 0.000s
```
so I'm nit sure if it makes sense to run the tests.

Reply:
I'm not sure why it fails if there is `$HOME` is unwriteable, but you can put a simple comment as `# no tests`. But, it's fairly common on python packages in NixOS to put this:

https://github.com/NixOS/nixpkgs/blob/644a7468b4a2a6b85f157b68986ea98f4d078760/doc/languages-frameworks/python.section.md#L635-L655

Reply:
OK, I added a comment.